python/insights_ansible_playbook_lib/_keygen.py

```diff
@@ -0,0 +1,214 @@
+import argparse
+import logging
+import os
+import pathlib
+import re
+import shutil
+import subprocess
+import sys
+import tempfile
+import traceback
+
+import insights_ansible_playbook_lib as lib
+from insights_ansible_playbook_lib.crypto import GPGCommandResult
+
+
+logger = logging.getLogger(__name__)
+
+
+if os.geteuid() == 0 and os.path.isdir("/var/lib/insights-ansible-playbook-verifier/"):
+    TEMPORARY_GPG_HOME_PARENT_DIRECTORY = "/var/lib/insights-ansible-playbook-verifier/"
+    TEMPORARY_GPG_HOME_PARENT_DIRECTORY_PREFIX = "gpg-"
+else:
+    TEMPORARY_GPG_HOME_PARENT_DIRECTORY = "/tmp/"
+    TEMPORARY_GPG_HOME_PARENT_DIRECTORY_PREFIX = (
+        "insights-ansible-playbook-verifier-gpg-"
+    )
```

Comment:
Since this is non-production code for mainly CI purposes, we can always use `/tmp/...`, instead of trying `/var/lib/`.

Non-blocking.

Reply:
Resolved